cpp/oneapi/dal/algo/decision_forest/detail/infer_ops.hpp

```diff
@@ -50,14 +58,41 @@ struct infer_ops {
                               const input_t& input,
                               const result_t& result) const {}
 
+    /// Check that the hyperparameters of the algorithm belong to the expected ranges
+    void check_parameters_ranges(const param_t& params, const input_t& input) const {
+        ONEDAL_ASSERT(params.get_block_size() > 0x0l);
```

Comment:
should this checks be not in ONEDAL_ASSERT, due to it works only with dbg flag? Its common question for all hp algorithms. Probably we can doublecheck in examples/tests

Reply:
Good point. Agree that those need to be modified to regular checks, not asserts.
But this change will have to be done in LinReg and Covariance also. I do not want to have all those changes part of this PR, rather will do it as a follow up work.